action.yaml

```diff
@@ -57,71 +57,82 @@ runs:
   steps:
     - name: Enable problem-matcher
       shell: bash
+      env:
+        format: ${{ inputs.format }}
+        disable_matcher: ${{ inputs.disable_matcher }}
```

Comment:
My point was that none of the inputs for this action make sense to have a dynamic based on PR/commit title/content, which is what the goal of this PR was.

BTW that recommendation is laughable, that is not true at all...
GitHub even hosted a CTF around it last year.

Reply:
> My point was that none of the inputs for this action make sense to have a dynamic based on PR/commit title/content, which is what the goal of this PR was.

One example that comes to mind is that someone might want to expose the `severity` input in such a way that someone writing a PR may opt-into running `shellcheck` in a more aggressive mode. This could be exposed by having a previous step read the PR title or description to extract a special value that is then passed into this action. For example, a previous step could look for `force_shellcheck_severity=...` in the description.

Another simpler approach would be to create a manual workflow that exposes the `severity` option as an input. This workflow could be used by someone who wants to allow devs to run shellcheck in a more aggressive mode in special cases.

These theoretical examples are likely not going to get implemented often but I think that there's reasonable enough that someone may want to something like that. In which case, they would be vulnerable to a script injection attack through this action.

Perhaps it's a question of philosophy. In my mind, when I use an action or a library, I expect it to be secure by default and to treat all inputs as untrusted. For example, if someone passes `$(echo hi)` as the `scandir` input, I expect it to look into a directory named `$(echo hi)` and fail. I don't expect it to actually run `echo hi`. Here I'm framing this as a security issue but it could also be seen as a regular bug because `$(echo hi)` is technically a valid directory name and it's not being handled correctly. The example here is convoluted but it's not completely impossible for someone to use `$(...)` somewhere in a directory name.